pkg/operator/ceph/file/filesystem_test.go

```diff
@@ -473,56 +473,81 @@ func TestUpgradeFilesystem(t *testing.T) {
 		},
 	}
 	createdFsResponse, _ := json.Marshal(mdsmap)
-	firstGet := false
+
+	// actual version
+	clusterInfo.CephVersion = version.Squid
+	// mocked version to cause an error different from the actual version
+	mockedVersionStr, _ := json.Marshal(
+		map[string]map[string]int{
+			"mds": {
+				"ceph version 18.2.0-0-g2f728b9 (2f728b952cf293dd7f809ad8a0f5b5d040c43010) reef (stable)": 2,
+			},
+		})
 	executor.MockExecuteCommandWithOutput = func(command string, args ...string) (string, error) {
-		if contains(args, "fs") && contains(args, "get") {
-			if firstGet {
-				firstGet = false
-				return "", errors.New("fs doesn't exist")
+		if contains(args, "fs") {
```

Comment:
While troubleshooting this unit test, I refactored a bit to consolidate similar commands for simpler debugging.